include/igl/writeMSH.cpp

```diff
@@ -0,0 +1,159 @@
+/* high level interface for MshSaver*/
+
+/* Copyright (C) 2020 Vladimir Fonov <vladimir.fonov@gmail.com> */
+/*
+/* This Source Code Form is subject to the terms of the Mozilla */
+/* Public License v. 2.0. If a copy of the MPL was not distributed */
+/* with this file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "writeMSH.h"
+#include "MshSaver.h"
+#include "MshLoader.h"
+#include <iostream>
+
+
+#if __cplusplus==201103L
+namespace igl {
+    // helper function, unfortunately impossible to make lambda in c++11
+    template <typename T,typename Derived> 
+    void __append_mat_to_vec(std::vector<T> &vec, const Eigen::PlainObjectBase<Derived> & mat) 
+    {
+        size_t st = vec.size();
+        vec.resize(st + mat.size());
+
+        Eigen::Map< Eigen::Matrix<typename Derived::Scalar, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> > 
+            _map_vec( reinterpret_cast<T *>( vec.data() + st ), mat.rows(), mat.cols() );
+        _map_vec = mat;
+    }
+
+}
+#endif
+
+
+IGL_INLINE bool igl::writeMSH(
+             const std::string    &msh,
+             const Eigen::MatrixXd &X,
+             const Eigen::MatrixXi &Tri,
+             const Eigen::MatrixXi &Tet,
+             const Eigen::MatrixXi &TriTag,
+             const Eigen::MatrixXi &TetTag,
+             const std::vector<std::string>     &XFields,
+             const std::vector<Eigen::MatrixXd> &XF,
+             const std::vector<std::string>     &EFields,
+             const std::vector<Eigen::MatrixXd> &TriF,
+             const std::vector<Eigen::MatrixXd> &TetF
+             )
+{
+    try 
+    {
+        #if __cplusplus>=201402L // c++14 have lambda with auto parameters
+        auto __append_mat_to_vec = [] (auto &vec, const auto & mat)-> void 
+        {
+            size_t st = vec.size();
+            vec.resize(st + mat.size());
+            using Scalar = decltype(mat.minCoeff()); // determine storage type of the matrix
+            Eigen::Map< Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> > 
+                _map_vec( reinterpret_cast<Scalar *>( vec.data() + st ), mat.rows(), mat.cols() );
+            _map_vec = mat;
+        };
+        #endif
```

Comment:
Not sure if lambda is more preferable over regular template functions even under c++14.